### ISSUE TITLE
[FEATURE] Support debug compilation for Rust with wash build

### ISSUE THREAD

**Issue by brooksmtownsend** [wasmCloud/wasmCloud#2298]
### Affected project(s)

- [ ] documentation
- [ ] examples
- [ ] wasmCloud host
- [X] wasmCloud CLI (wash)
- [ ] wasmCloud dashboard UI (washboard)
- [X] capability providers
- [ ] provider bindgen
- [ ] control interface client
- [ ] other / not sure

### Is your feature request related to a problem? Please describe.

Building release mode binaries takes a significantly longer time than building debug mode binaries for Rust. When rapidly iterating on a capability provider, always having to build in release mode slows down the feedback loop.

### Describe the solution you'd like

I would love to be able to specify in my `wasmcloud.toml` file to build for debug mode instead of release for Rust providers. For example:
```toml
name = "Messaging NATS"
language = "rust"
type = "provider"
version = "0.20.0"

[rust]
target_dir = "../../"
# Added field
debug = true

[provider]
bin_name = "messaging-nats-provider"
vendor = "wasmCloud"
```

This `debug` field could default to `false` and when it's false build for release. We could use this for components as well.

### Describe alternatives you've considered

We could make debug the default, but that would be a change in behavior and I'd rather `wash build` prepare an optimized binary by default. 

### Additional context

For those looking to contribute, here's where you may want to look for making these changes:

To build components: https://github.com/wasmCloud/wasmCloud/blob/main/crates/wash-lib/src/build/component.rs#L187

To build providers: https://github.com/wasmCloud/wasmCloud/blob/main/crates/wash-lib/src/build/provider.rs#L108

**Comment by Alex-omosa** [comment]
@brooksmtownsend I would like to pick this one up.

**Comment by brooksmtownsend** [comment]
Sure thing @Alex-omosa ! Looking forward to the PR 😄

**Comment by vados-cosmonic** [comment]
Hey @Alex-omosa were you able to get to this by any chance?

**Comment by Alex-omosa** [comment]
Hey @vados-cosmonic haven't gotten a chance yet. I'll work on it this week.

**Comment by vados-cosmonic** [comment]
Awesome @Alex-omosa thanks :) Please feel free to reach out if you need anything or early reviews, etc -- we're always glad to help (it can be cumbersome knowing where to start in the codebase!)

**Comment by Alex-omosa** [comment]
Thanks @vados-cosmonic. I appreciate your help. I'll reach out when I need anything, so far I'm good.
(I'm completely new to rust btw 😂).

**Comment by Alex-omosa** [comment]
Hi @vados-cosmonic. Kindly review this [PR](https://github.com/wasmCloud/wasmCloud/pull/2446) for me.

**Comment by vados-cosmonic** [comment]
Hey @Alex-omosa thanks! I'll take a look :)

**Comment by vados-cosmonic** [comment]
Resolved in #2446
